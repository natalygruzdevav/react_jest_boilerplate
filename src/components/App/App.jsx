import './App.css';

import React, { Component } from 'react';
import Tabs from '../Tabs';

class App extends Component {
  render() {
    return (
      <div>
        <h1>My React App!</h1>
        <Tabs />
      </div>
    );
  }
}

export default App;
