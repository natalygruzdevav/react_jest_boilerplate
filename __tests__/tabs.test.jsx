import React from 'react';
import { mount } from 'enzyme';

import App from '../src/components/App';

describe('Tabs', () => {
  it('render', () => {
    const wrapper = mount(<App />);
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('click tab', () => {
    const wrapper = mount(<App />);
    wrapper
      .find('[data-test="tab-item"]')
      .at(4)
      .simulate('click');
    expect(wrapper.render()).toMatchSnapshot();
  });
});
