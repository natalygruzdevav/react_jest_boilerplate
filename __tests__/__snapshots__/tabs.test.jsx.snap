// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs click tab 1`] = `
<div>
  <h1>
    My React App!
  </h1>
  <div
    class="react-tabs"
    data-tabs="true"
  >
    <ul
      class="react-tabs__tab-list"
      role="tablist"
    >
      <li
        aria-controls="react-tabs-13"
        aria-disabled="false"
        aria-selected="false"
        class="react-tabs__tab"
        data-test="tab-item"
        id="react-tabs-12"
        role="tab"
      >
        Title 1
      </li>
      <li
        aria-controls="react-tabs-15"
        aria-disabled="false"
        aria-selected="false"
        class="react-tabs__tab"
        data-test="tab-item"
        id="react-tabs-14"
        role="tab"
      >
        Title 2
      </li>
      <li
        aria-controls="react-tabs-17"
        aria-disabled="false"
        aria-selected="true"
        class="react-tabs__tab react-tabs__tab--selected"
        data-test="tab-item"
        id="react-tabs-16"
        role="tab"
        tabindex="0"
      >
        Title 3
      </li>
      <li
        aria-controls="react-tabs-19"
        aria-disabled="false"
        aria-selected="false"
        class="react-tabs__tab"
        data-test="tab-item"
        id="react-tabs-18"
        role="tab"
      >
        Title 4
      </li>
      <li
        aria-controls="react-tabs-21"
        aria-disabled="false"
        aria-selected="false"
        class="react-tabs__tab"
        data-test="tab-item"
        id="react-tabs-20"
        role="tab"
      >
        Title 5
      </li>
      <li
        aria-controls="react-tabs-23"
        aria-disabled="false"
        aria-selected="false"
        class="react-tabs__tab"
        data-test="tab-item"
        id="react-tabs-22"
        role="tab"
      >
        Title 6
      </li>
    </ul>
    <div
      aria-labelledby="react-tabs-12"
      class="react-tabs__tab-panel"
      id="react-tabs-13"
      role="tabpanel"
    />
    <div
      aria-labelledby="react-tabs-14"
      class="react-tabs__tab-panel"
      id="react-tabs-15"
      role="tabpanel"
    />
    <div
      aria-labelledby="react-tabs-16"
      class="react-tabs__tab-panel react-tabs__tab-panel--selected"
      id="react-tabs-17"
      role="tabpanel"
    >
      <h2>
        Any content 3
      </h2>
    </div>
    <div
      aria-labelledby="react-tabs-18"
      class="react-tabs__tab-panel"
      id="react-tabs-19"
      role="tabpanel"
    />
    <div
      aria-labelledby="react-tabs-20"
      class="react-tabs__tab-panel"
      id="react-tabs-21"
      role="tabpanel"
    />
    <div
      aria-labelledby="react-tabs-22"
      class="react-tabs__tab-panel"
      id="react-tabs-23"
      role="tabpanel"
    />
  </div>
</div>
`;

exports[`Tabs render 1`] = `
<div>
  <h1>
    My React App!
  </h1>
  <div
    class="react-tabs"
    data-tabs="true"
  >
    <ul
      class="react-tabs__tab-list"
      role="tablist"
    >
      <li
        aria-controls="react-tabs-1"
        aria-disabled="false"
        aria-selected="true"
        class="react-tabs__tab react-tabs__tab--selected"
        data-test="tab-item"
        id="react-tabs-0"
        role="tab"
        tabindex="0"
      >
        Title 1
      </li>
      <li
        aria-controls="react-tabs-3"
        aria-disabled="false"
        aria-selected="false"
        class="react-tabs__tab"
        data-test="tab-item"
        id="react-tabs-2"
        role="tab"
      >
        Title 2
      </li>
      <li
        aria-controls="react-tabs-5"
        aria-disabled="false"
        aria-selected="false"
        class="react-tabs__tab"
        data-test="tab-item"
        id="react-tabs-4"
        role="tab"
      >
        Title 3
      </li>
      <li
        aria-controls="react-tabs-7"
        aria-disabled="false"
        aria-selected="false"
        class="react-tabs__tab"
        data-test="tab-item"
        id="react-tabs-6"
        role="tab"
      >
        Title 4
      </li>
      <li
        aria-controls="react-tabs-9"
        aria-disabled="false"
        aria-selected="false"
        class="react-tabs__tab"
        data-test="tab-item"
        id="react-tabs-8"
        role="tab"
      >
        Title 5
      </li>
      <li
        aria-controls="react-tabs-11"
        aria-disabled="false"
        aria-selected="false"
        class="react-tabs__tab"
        data-test="tab-item"
        id="react-tabs-10"
        role="tab"
      >
        Title 6
      </li>
    </ul>
    <div
      aria-labelledby="react-tabs-0"
      class="react-tabs__tab-panel react-tabs__tab-panel--selected"
      id="react-tabs-1"
      role="tabpanel"
    >
      <h2>
        Any content 1
      </h2>
    </div>
    <div
      aria-labelledby="react-tabs-2"
      class="react-tabs__tab-panel"
      id="react-tabs-3"
      role="tabpanel"
    />
    <div
      aria-labelledby="react-tabs-4"
      class="react-tabs__tab-panel"
      id="react-tabs-5"
      role="tabpanel"
    />
    <div
      aria-labelledby="react-tabs-6"
      class="react-tabs__tab-panel"
      id="react-tabs-7"
      role="tabpanel"
    />
    <div
      aria-labelledby="react-tabs-8"
      class="react-tabs__tab-panel"
      id="react-tabs-9"
      role="tabpanel"
    />
    <div
      aria-labelledby="react-tabs-10"
      class="react-tabs__tab-panel"
      id="react-tabs-11"
      role="tabpanel"
    />
  </div>
</div>
`;
